packages/playwright-core/src/server/injected/roleUtils.ts

```diff
@@ -1013,18 +1013,19 @@ function belongsToDisabledFieldSet(element: Element): boolean {
   return !legendElement || !legendElement.contains(element);
 }
 
-function hasExplicitAriaDisabled(element: Element | undefined): boolean {
+function hasExplicitAriaDisabled(element: Element | undefined, isAncestor = false): boolean {
   if (!element)
     return false;
-  if (kAriaDisabledRoles.includes(getAriaRole(element) || '')) {
+  if (isAncestor || kAriaDisabledRoles.includes(getAriaRole(element) || '')) {
```

Comment:
According to https://www.w3.org/TR/wai-aria-1.2/#aria-disabled, aria disabled only applies to elements with a suitable role, when that element or one of the ancestors (no matter the ancestor's role) has `aria-disabled=true`.

Double checking with DevTools accessibility panel:
- In the following snippet `<h1>` is not disabled, because it has a `heading` role:
  ```
  <span aria-disabled="true"> <h1>Click me!</h1> </span>
  ```

- In the following snippet `<button>` is disabled, because it has a suitable `button` role:
  ```
  <span aria-disabled="true"> <button>Click me!</button> </span>
  ```

Reply:
Why are you saying it all?

Reply:
Are you sure this is the expected behavior? Our test started to fail after this change with following test code and  html:


```
await expect(page.getByText('name1')).toBeDisabled()
```

```
...
<div aria-disabled="true" role="option">
  <span class="name">name1</span>
   <span class="description">description1</span>
</div>
...
```

The reason for the failure is that `getByText('name1')` returns a span element which has no suitable role in `kAriaDisabledRoles`. For me it doesn't make sense to add `role="option"`  also for the span element.